secure/storage/src/tests/suite.rs

```diff
@@ -150,48 +127,15 @@ fn test_verify_incorrect_value_types_unwrap(storage: &mut dyn Storage) {
     );
 }
 
-/// This test attempts to create a key/value pair twice, asserting that the second
-/// creation attempt fails and returns a key exists error.
-fn test_create_key_value_twice(storage: &mut dyn Storage) {
-    let u64_value = 10;
-    let policy = Policy::public();
-
-    assert_eq!(
-        storage.create(U64_KEY, Value::U64(u64_value), &policy),
-        Ok(())
-    );
-    match storage.create(U64_KEY, Value::U64(u64_value), &policy) {
-        Err(Error::KeyAlreadyExists(_)) => (/* Expected error code */),
-        _ => panic!("The second call to create() should have failed!"),
-    }
-}
-
 /// This test: (i) creates a new named test key pair; (ii) retrieves the public key for
 /// the created key pair; (iii) compares the public keys returned by the create call and the
 /// retrieval call.
-fn test_create_get_and_test_key_pair(storage: &mut dyn Storage) {
-    let public_key = storage
-        .create_key(CRYPTO_NAME, &Policy::public())
-        .expect("Failed to create a test Ed25519 key pair!");
-    let retrieved_public_key_response = storage
-        .get_public_key(CRYPTO_NAME)
-        .expect("Failed to fetch the test key pair!");
+fn test_create_get_key_pair(storage: &mut dyn Storage) {
+    let public_key = storage.create_key(CRYPTO_NAME, &Policy::public()).unwrap();
+    let retrieved_public_key_response = storage.get_public_key(CRYPTO_NAME).unwrap();
     assert_eq!(public_key, retrieved_public_key_response.public_key);
 }
 
-/// This test attempts to create two named key pairs using the same name, and asserts
```

Comment:
no, there is no guarantee about this... create_key on a KVStorage will likely allow multiple calls. Doesn't seem reasonable to enforce such an arbitrary constraint.

Reply:
My only push back in this PR is removing this test given that the handling of create_key() and policies hasn't been addressed (it's something you mentioned doing in future work..). I can see why the test now fails though (because internally create_key() has now been updated to use set() rather than create()), but it seems like we could update the implementation to maintain the original semantics (e.g., do a check before set). Anyway, it's not a big deal -- just might be a little cleaner to remove it in the PR that handles create_key()

Reply:
But then we'd have to change the semantic of create_key within the crypto kv to support some arbitrary behavior. If you asked me a few weeks ago, I'd would agree, but I think now we need to be lean with incrementalism unless we are certain we want a specific behavior.